example_site/src/components/Collections.js

```diff
@@ -31,45 +32,124 @@ function Template({ template }) {
   );
 }
 
+function DurationFilter({
+  filter,
+  templateFilter,
+  setTemplateFilter,
+}) {
+  const [buttonStatus, setButtonStatus] = useState(true);
+  const buttonColor = templateFilter.duration === filter.value ? theBlue : 'black';
+
+  const onSelectFilter = (newDuration) => {
+    if (buttonStatus) {
+      setTemplateFilter((currentFilter) => (
+        {...currentFilter, duration: newDuration.value}
+      ))  
+    } else {
+      setTemplateFilter((currentFilter) => {
+        const newFilters = {...currentFilter};
+        delete newFilters.duration;
+        return newFilters;
+      })
+    }
+  }
+
+  return (
+    <div className="category-name">
+      <button 
+        className="filter-name"
+        onClick={() => {
+          onSelectFilter(filter)
+          setButtonStatus(!buttonStatus)
+      }}>
+        <div style={{'color': buttonColor}}>
+          {filter.displayName} 
+        </div>
+      </button>
+    </div>
+  );
+}
+
+function AspectRatioFilter({
+  filter,
+  templateFilter,
+  setTemplateFilter,
+}) {
+  const [buttonStatus, setButtonStatus] = useState(true);
+  const buttonColor = templateFilter.aspectRatio === filter.value ? theBlue : 'black';
+
+  const onSelectFilter = (newAspectRatio) => {
+    if (buttonStatus) {
+      setTemplateFilter((currentFilter) => (
+        {...currentFilter, aspectRatio: newAspectRatio.value}
+      ))  
+    } else {
+      setTemplateFilter((currentFilter) => {
+        const newFilters = {...currentFilter};
+        delete newFilters.aspectRatio;
+        return newFilters;
+      })
+    }
+  }
+
+  return (
+    <div className="category-name">
+      <button
+        className="filter-name"
+        onClick={() => {
+          onSelectFilter(filter)
+          setButtonStatus(!buttonStatus)
+        }}>
+        <div style={{'color': buttonColor}}>
+          {filter.displayName} 
+        </div>
+      </button>
+    </div>
+  );
+}
+
 function CollectionNames({
+  collections,
   collection,
+  selectedCollection,
   setSelectedCollection,
 }) {
-  const { waymarkInstance, addTemplates } = useAppContext();
+  const buttonColor = selectedCollection === collection ? theBlue : 'black';
 
   const onClick = (collection) => {
     setSelectedCollection(collection);
   };
 
+  if (!selectedCollection) {
+    setSelectedCollection(collections.find((collection) => (
```

Comment:
What do you think of keeping this logic in the top level component and always passing in a selected collection? The only time there should not be one is when the component loads for the first time, so we should be able to put this logic in selected collection state initialization.